app/lib/dfe/wizard.rb

```diff
@@ -0,0 +1,138 @@
+module DfE
+  class Wizard
+    attr_reader :current_step_name, :steps, :request
+    attr_writer :step_params
+
+    delegate :next_step, to: :current_step
+    delegate :info, to: :logger, allow_nil: true
+
+    def initialize(current_step:, step_params: {}, **args)
+      @current_step_name = current_step
+      @step_params = step_params
+      @steps = self.class.steps
+
+      (args || {}).each do |key, value|
+        send("#{key}=", value) if respond_to?("#{key}=")
+      end
+    end
+
+    def self.steps
+      return @steps unless block_given?
+
+      @steps = yield
+    end
+
+    def self.store(store_service = nil)
+      @store ||= store_service
+    end
+
+    def logger; end
+
+    def current_step
+      instance_current_step if step_form_object_class.present?
+    end
+
+    def valid_step?
+      current_step.valid?
+    end
+
+    def invalid_step?
+      !valid_step?
+    end
+
+    def save
+      store.save if store.present?
+    end
+
+    def store
+      @store ||= self.class.store.new(self) if self.class.store.present?
+    end
+
+    def step_params
+      if @step_params.respond_to?(:permit) && @step_params.key?(current_step_name)
```

Comment:
The reasoning behind this method appears to relate to the fact that the type of the object varies between `ActionController::Parameters` and a `Hash`. Could things be made simpler by guaranteeing that the object is `AC::P`?